### PR TITLE
[SDN-1364] Update OVN-K config for Audit logging

### PR DESCRIPTION
**Pull request by astoycos**
API changes For [SDN-1364](https://issues.redhat.com/browse/SDN-1364)

See [Enhancement](https://github.com/openshift/enhancements/pull/617) for
details on the changes

Signed-off-by: Andrew Stoycos <astoycos@redhat.com>

### CODE REVIEW

File: operator/v1/types_network.go [hunk 1]
```diff
@@ -349,6 +353,35 @@ type HybridOverlayConfig struct {
 type IPsecConfig struct {
 }
 
+type PolicyAuditConfig struct {
+	// RateLimit is the approximate maximum number of messages to generate per-second per-node. If
```
Comment (sttts) [inline operator/v1/types_network.go:357]: lower-case all field names in the description. They relate to the JSON representation.

File: operator/v1/types_network.go [hunk 2]
```diff
@@ -349,6 +353,35 @@ type HybridOverlayConfig struct {
...
+	// unset the default of 20 msg/sec is used.
+	// +optional
+	RateLimit *uint32 `json:"rateLimit,omitempty"`
+
+	// MaxFilesSize is the max size an ACL_audit log file is allowed to reach before rotation occurs
+	// Default is 50MB
+	// +optional
+	MaxFileSize *uint32 `json:"maxFileSize,omitempty"`
```
Comment (sttts) [inline operator/v1/types_network.go:365]: which unit? So default is 50000000 ? Are you sure about uint32? 

Is zero a valid file size?

File: operator/v1/types_network.go [hunk 3]
```diff
@@ -349,6 +353,35 @@ type HybridOverlayConfig struct {
 type IPsecConfig struct {
 }
 
+type PolicyAuditConfig struct {
+	// RateLimit is the approximate maximum number of messages to generate per-second per-node. If
+	// unset the default of 20 msg/sec is used.
+	// +optional
+	RateLimit *uint32 `json:"rateLimit,omitempty"`
```
Comment (sttts) [inline operator/v1/types_network.go:360]: Is zero a valid rate limit?

File: operator/v1/types_network.go [hunk 4]
```diff
@@ -349,6 +353,35 @@ type HybridOverlayConfig struct {
...
+
+	// Messages are output in syslog format. Destination is the destination for policy log messages.
+	// Regardless of this config logs will always be dumped to ovn at /var/log/ovn/ however
+	// you may also configure additional output as follows.
+	// Messages are output in syslog format.
+	// Valid values are:
+	// - "libc" -> to use the libc syslog() function of the host node's journdald process
+	// - "udp:host:port" -> for sending syslog over UDP
+	// - "unix:file" -> for using the UNIX domain socket directly
+	// - "null" -> to discard all messages logged to syslog
```
Comment (sttts) [inline operator/v1/types_network.go:375]: add validation for all of them (regex)

File: operator/v1/types_network.go [hunk 5]
```diff
@@ -349,6 +353,35 @@ type HybridOverlayConfig struct {
...
+	// The default is "null"
```
Comment (sttts) [inline operator/v1/types_network.go:376]: worth to default this explicitly as a value?

File: operator/v1/types_network.go [hunk 6]
```diff
@@ -349,6 +353,35 @@ type HybridOverlayConfig struct {
...
+	// +optional
+	Destination string `json:"destination,omitempty"`
+
+	// SyslogFacility the RFC5424 facility for generated messages, e.g. "kern". Default is "local0"
+	// +optional
+	SyslogFacility string `json:"syslogFacility,omitempty"`
```
Comment (sttts) [inline operator/v1/types_network.go:382]: as these are known, please add an enum validation and use an explicit default.

Comment (sttts) [reply operator/v1/types_network.go:382]: Make it required then of course.

File: operator/v1/types_network.go (hunk 1 above)
Comment (astoycos) [reply operator/v1/types_network.go:357]: Sounds good

File: operator/v1/types_network.go (hunk 2 above)
Comment (astoycos) [reply operator/v1/types_network.go:365]: I think I will Try to change the unit here to MB, 0 should not be valid

File: operator/v1/types_network.go (hunk 3 above)
Comment (astoycos) [reply operator/v1/types_network.go:360]: No

File: operator/v1/types_network.go (hunk 4 above)
Comment (astoycos) [reply operator/v1/types_network.go:375]: Can't I perform validation for this field in the Cluster network operator @sttts? That's what I was planning on doing anyways.

File: operator/v1/types_network.go (hunk 5 above)
Comment (astoycos) [reply operator/v1/types_network.go:376]: The Defaults are populated here -> https://github.com/openshift/cluster-network-operator/blob/master/pkg/network/ovn_kubernetes.go#L246

File: operator/v1/types_network.go (hunk 6 above)
Comment (astoycos) [reply operator/v1/types_network.go:382]: This is optional, and only configured if the syslog `Destination` is set to something besides "null", again default is set by CNO

File: operator/v1/types_network.go (hunk 4 above)
Comment (astoycos) [reply operator/v1/types_network.go:379]: Added Regex Validation and Defaults

File: operator/v1/types_network.go (hunk 5 above)
Comment (astoycos) [reply operator/v1/types_network.go:380]: Also Added Defaults here as well

File: operator/v1/types_network.go (hunk 6 above)
Comment (astoycos) [reply operator/v1/types_network.go:390]: Add regex validation, Let me know if this works, and thanks for the review @sttts!

**Comment by knobunc** [comment]
/lgtm

@sttts are you okay with this based on @astoycos's responses?  Or do you want changes?

**Pull request merged**